model/messages/internal.go

```diff
@@ -0,0 +1,88 @@
+package messages
+
+import (
+	"context"
+
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/trace"
+	"go.opentelemetry.io/otel/label"
+
+	"github.com/filecoin-project/sentinel-visor/metrics"
+	"github.com/filecoin-project/sentinel-visor/model"
+)
+
+type InternalMessage struct {
+	tableName     struct{} `pg:"internal_messages"` // nolint: structcheck,unused
+	Height        int64    `pg:",pk,notnull,use_zero"`
+	Cid           string   `pg:",pk,notnull"`
+	StateRoot     string   `pg:",notnull"`
+	SourceMessage string
+	From          string `pg:",notnull"`
+	To            string `pg:",notnull"`
+	Value         string `pg:"type:numeric,notnull"`
+	Method        uint64 `pg:",use_zero"`
+	ActorName     string `pg:",notnull"`
+	ActorFamily   string `pg:",notnull"`
+	ExitCode      int64  `pg:",use_zero"`
+	GasUsed       int64  `pg:",use_zero"`
+}
+
+func (im *InternalMessage) Persist(ctx context.Context, s model.StorageBatch, version model.Version) error {
+	ctx, _ = tag.New(ctx, tag.Upsert(metrics.Table, "internal_messages"))
+	stop := metrics.Timer(ctx, metrics.PersistDuration)
+	defer stop()
+
+	return s.PersistModel(ctx, im)
+}
+
+type InternalMessageList []*InternalMessage
+
+func (l InternalMessageList) Persist(ctx context.Context, s model.StorageBatch, version model.Version) error {
+	if len(l) == 0 {
+		return nil
+	}
+	ctx, span := global.Tracer("").Start(ctx, "InternalMessageList.Persist", trace.WithAttributes(label.Int("count", len(l))))
+	defer span.End()
+
+	ctx, _ = tag.New(ctx, tag.Upsert(metrics.Table, "internal_messages"))
+	stop := metrics.Timer(ctx, metrics.PersistDuration)
+	defer stop()
+
+	return s.PersistModel(ctx, l)
+}
+
+type InternalParsedMessage struct {
+	tableName struct{} `pg:"internal_parsed_messages"` // nolint: structcheck,unused
+	Height    int64    `pg:",pk,notnull,use_zero"`
+	Cid       string   `pg:",pk,notnull"`
+	From      string   `pg:",notnull"`
+	To        string   `pg:",notnull"`
+	Value     string   `pg:"type:numeric,notnull"`
+	Method    uint64   `pg:",use_zero"`
```

Comment:
should be of type string based on comment here: https://github.com/filecoin-project/sentinel-visor/issues/482#issuecomment-853867484

Reply:
good catch, will fix

Reply:
I've copy-pasted these structs (I fixed this bit) into my internal_message branch: https://github.com/filecoin-project/sentinel-visor/pull/533. Are you alright with #533 landing as a follow-on to this?

Reply:
Yep, let's do it as a follow-up